webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable react/jsx-no-target-blank */
+import React from "react";
+import { OnboardingStep } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects, NotFoundError } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_OBJECT_PATH = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  try {
+    await objects.head(repo, defaultBranch, SAMPLE_REPO_OBJECT_PATH);
+    return true;
+  } catch (e) {
+    return false;
+  }
+  
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const promises = [
+    retention.setGCPolicyPreflight(repoId),
+    branchProtectionRules.createRulePreflight(repoId),
+    objects.uploadPreflight(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    ),
```

Comment:
Let's get rid of the random filename!  We could just use
```suggestion
    objects.uploadPreflight(
      repoId,
      defaultBranch,
      `${HOOKS_PATH}sample_hook`
    ),
```
or any other constant string.  It's not like we plan to detect on only _some_ files being allowed under HOOKS_PATH, just that _something_ can be written there.  That way we might rid of the requirement for an ugly "random" filename

Reply:
Done